go-controller/pkg/ovn/hybrid.go

```diff
@@ -1,151 +1,55 @@
-			}
-			return m.DeleteNode(node)
-		},
-		informer.ReceiveAllUpdates,
-	)
-
-	// Add our hybrid overlay CIDRs to the subnetallocator
-	for _, clusterEntry := range config.HybridOverlay.ClusterSubnets {
-		err := m.allocator.AddNetworkRange(clusterEntry.CIDR, clusterEntry.HostSubnetLength)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	// Mark existing hostsubnets as already allocated
-	existingNodes, err := m.kube.GetNodes()
-	if err != nil {
-		return nil, fmt.Errorf("error in initializing/fetching subnets: %v", err)
-	}
-	for _, node := range existingNodes.Items {
-		hostsubnet, err := houtil.ParseHybridOverlayHostSubnet(&node)
-		if err != nil {
-			klog.Warningf(err.Error())
-		} else if hostsubnet != nil {
-			klog.V(5).Infof("Marking existing node %s hybrid overlay NodeSubnet %s as allocated", node.Name, hostsubnet)
-			if err := m.allocator.MarkAllocatedNetwork(hostsubnet); err != nil {
-				utilruntime.HandleError(err)
-			}
-		}
-	}
-
-	return m, nil
-}
-
-// Run starts the controller
-func (m *MasterController) Run(stopCh <-chan struct{}) {
-	defer utilruntime.HandleCrash()
-	klog.Info("Starting Hybrid Overlay Master Controller")
-
-	wg := &sync.WaitGroup{}
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		err := m.nodeEventHandler.Run(informer.DefaultNodeInformerThreadiness, stopCh)
-		if err != nil {
-			klog.Error(err)
-		}
-	}()
-	<-stopCh
-	klog.Info("Shutting down Hybrid Overlay Master workers")
-	wg.Wait()
-	klog.Info("Shut down Hybrid Overlay Master workers")
-}
-
 // hybridOverlayNodeEnsureSubnet allocates a subnet and sets the
 // hybrid overlay subnet annotation. It returns any newly allocated subnet
 // or an error. If an error occurs, the newly allocated subnet will be released.
-func (m *MasterController) hybridOverlayNodeEnsureSubnet(node *kapi.Node, annotator kube.Annotator) (*net.IPNet, error) {
+func (oc *Controller) hybridOverlayNodeEnsureSubnet(node *kapi.Node, annotator kube.Annotator) (*net.IPNet, error) {
 	// Do not allocate a subnet if the node already has one
 	if subnet, _ := houtil.ParseHybridOverlayHostSubnet(node); subnet != nil {
 		return nil, nil
 	}
 
 	// Allocate a new host subnet for this node
-	hostsubnets, err := m.allocator.AllocateNetworks()
+	hostsubnets, err := oc.hybridOverlaySubnetAllocator.AllocateNetworks()
 	if err != nil {
 		return nil, fmt.Errorf("error allocating hybrid overlay HostSubnet for node %s: %v", node.Name, err)
 	}
 
 	if err := annotator.Set(types.HybridOverlayNodeSubnet, hostsubnets[0].String()); err != nil {
-		_ = m.allocator.ReleaseNetwork(hostsubnets[0])
+		_ = oc.hybridOverlaySubnetAllocator.ReleaseNetwork(hostsubnets[0])
 		return nil, err
 	}
 
 	klog.Infof("Allocated hybrid overlay HostSubnet %s for node %s", hostsubnets[0], node.Name)
 	return hostsubnets[0], nil
 }
 
-func (m *MasterController) releaseNodeSubnet(nodeName string, subnet *net.IPNet) error {
+func (oc *Controller) releaseNodeSubnet(nodeName string, subnet *net.IPNet) error {
```

Comment:
@JacobTanenbaum can we rename this to `releaseHybridOverlaySubnet()` ?

Reply:
@dcbw I renamed it to `relaseHybridOverlayNodeSubnet()` I thought keeping the word Node in the name made it clearer that the function removes the subnet for the node. does that make sense? does that change meet the spirit of your comment or do you prefer `releaseHybridOverlaySubnet()` much better?

Reply:
Sounds fine to me.